packages/indexer/src/messaging/priceWorker.ts

```diff
@@ -0,0 +1,242 @@
+import Redis from "ioredis";
+import winston from "winston";
+import { Job, Worker } from "bullmq";
+import { DataSource, entities } from "@repo/indexer-database";
+import { IndexerQueues } from "./service";
+import { ethers } from "ethers";
+import { yesterday, findTokenByAddress } from "../utils";
+import { CoingeckoClient } from "../utils/coingeckoClient";
+import { RetryProvidersFactory } from "../web3/RetryProvidersFactory";
+import { assert } from "@repo/error-handling";
+
+export type PriceMessage = {
+  depositId: number;
+  originChainId: number;
+};
+
+/**
+ * This worker listens to the `PriceQuery` queue and processes each job by:
+ * - Retrieving the deposit and relay hash information from the database using the deposit ID and origin chain ID.
+ * - Verifying the existence of the relay hash info and deposit records.
+ * - Determining the block time from the relay hash info and calculating the price time as the previous day's timestamp.
+ * - Identifying the base currency using the output token and destination chain ID.
+ * - Checking if a historic price for the base currency and quote currency (USD) already exists in the database.
+ * - If not, fetching the historic price from Coingecko and inserting it into the database.
+ * - Logging errors and information at various stages of the process.
+ */
+export class PriceWorker {
+  private worker: Worker;
+  private coingeckoClient: CoingeckoClient;
+  private relayHashInfoRepository;
+  private depositRepository;
+  private historicPriceRepository;
+
+  constructor(
+    private redis: Redis,
+    private postgres: DataSource,
+    private logger: winston.Logger,
+  ) {
+    this.coingeckoClient = new CoingeckoClient();
+    this.relayHashInfoRepository = this.postgres.getRepository(
+      entities.RelayHashInfo,
+    );
+    this.depositRepository = this.postgres.getRepository(
+      entities.V3FundsDeposited,
+    );
+    this.historicPriceRepository = this.postgres.getRepository(
+      entities.HistoricPrice,
+    );
+    this.setWorker();
+  }
+  private async getPrice(
+    address: string,
+    chainId: number,
+    time: Date,
+    quoteCurrency = "usd",
+  ): Promise<number> {
+    const priceTime = yesterday(time);
+    const tokenInfo = findTokenByAddress(address, chainId);
+    const baseCurrency = tokenInfo.symbol;
+
+    const cachedPrice = await this.historicPriceRepository.findOne({
+      where: {
+        date: priceTime,
+        baseCurrency,
+        quoteCurrency,
+      },
+    });
+    // we have this price at this time in the db
+    if (cachedPrice) return Number(cachedPrice.price);
```

Comment:
nit: w/out the braces it's not immediately obvious that this isn't a one-liner but rather a conditional + a return
```suggestion
    if (cachedPrice) { 
        return Number(cachedPrice.price);
    }
```